libvmi/read.c

```diff
@@ -97,13 +97,13 @@ vmi_read(
 
         if(dtb) {
             paddr = vmi_pagetable_lookup(vmi, dtb, start_addr + buf_offset);
+            if (!paddr) {
```

Comment:
In that case we might need to extend the extended lookup to also return an error code for failed translations (when PTE ENTRY_PRESENT() is not true)